### ISSUE TITLE
MissingPackageManifestError: Could not find one of 'package.json' manifest files in the package

### ISSUE THREAD

**Issue by mbateman66** [esphome/issues#6324]
### The problem

Fresh install of HA and ESPHome on RaspPi. HA seems to be working fine. I used ESPHome web to flash a new ESP-WROOM-32. That all worked, and I was able to connect to my wifi and adopt the device. Without editing anything in the config yaml file. I tried to do an install, and got the following:

`INFO ESPHome 2024.9.2
INFO Reading configuration /config/esphome/esphome-web-5e22d0.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing esphome-web-5e22d0 (board: esp32dev; framework: arduino; platform: platformio/espressif32@5.4.0)
MissingPackageManifestError: Could not find one of 'package.json' manifest files in the package`

### Which version of ESPHome has the issue?

2024.9.2

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2024.10.2

### What platform are you using?

ESP32

### Board

ESP-WROOM-32

### Component causing the issue

install

### Example YAML snippet

```yaml
substitutions:
  name: esphome-web-5e22d0
  friendly_name: Mailbox Motion

esphome:
  name: ${name}
  friendly_name: ${friendly_name}
  min_version: 2024.6.0
  name_add_mac_suffix: false
  project:
    name: esphome.web
    version: dev

esp32:
  board: esp32dev
  framework:
    type: arduino
    

# Enable logging
logger:

# Enable Home Assistant API
api:

# Allow Over-The-Air updates
ota:
- platform: esphome

# Allow provisioning Wi-Fi via serial
improv_serial:

wifi:
  # Set up a wifi access point
  ap: {}

# In combination with the `ap` this allows the user
# to provision wifi credentials to the device via WiFi AP.
captive_portal:

dashboard_import:
  package_import_url: github://esphome/example-configs/esphome-web/esp32.yaml@main
  import_full_config: true

# Sets up Bluetooth LE (Only on ESP32) to allow the user
# to provision wifi credentials to the device.
esp32_improv:
  authorizer: none

# To have a "next url" for improv serial
web_server:
```

### Additional information

I have tried uninstalling and reinstalling ESPHome add-on a couple of times with no change in behavior.

**Comment by mbateman66** [comment]
FYI, I uninstalled ESPHome from HA and installed the Beta version (2024.10.0b2). This one works, so the issue seems to be in the main release version.

**Comment by mbateman66** [comment]
This seems to be fixed in 2024.10.0
